### DISCUSSION TITLE
Request for Simple Text Input

### DISCUSSION THREAD

**Discussion by xloc** [zauberzeug/nicegui#525] [Ideas / Feature Requests]
Hi there,

Thank you so much for bringing us such a nice UI library! I love it!

I wonder if we have a simpler text input alternative to the current one. The current input has too many design elements and isn't customizable enough like other UI elements. The textfield is deeply nested, which makes it difficult to apply styles using tailwind-css.

Would it be possible to add a simple text input component with no associated label or nested element, so that we can easily apply styles like we can with other components? What's your thought?

**Comment by falkoschindler** [comment]
You're right, Quasar's QInput element (which `ui.input` is building upon) is a bit tricky to style. But starting again from a plain HTML input would somehow re-invent the wheel. So we are not planning to introduce a dedicated NiceGUI element doing this. Instead we try to simplify styling in general (#117, #362, #364, ...).

But if you really want a plain input tag, you can always use the generic `ui.element` to do so:

```py
ui.element('input').props('value="Hello world!"') \
    .on('input', lambda msg: ui.notify(msg['args']['target']['_value']), args=['target'])
```

The event registration turned out a bit challenging, but it works. Basically we need to request the "target" field of the JavaScript event and navigate to the "_value".

**Comment by xloc** [comment]
Okay, understood. Thanks for your time to provide the walk around! Let me try it out
